### PR TITLE
Add support for registering custom loss functions for Keras import

### PR DESCRIPTION
**Pull request by treo**
This adds support for registering custom loss functions for models imported from Keras.

This should fix issues like:
https://github.com/eclipse/deeplearning4j/issues/7895

### CODE REVIEW

**Review by AlexDBlack — APPROVED** [review]
LGTM, thanks!

There's 1 failure in CI unrelated to your changes that I've logged here: https://github.com/eclipse/deeplearning4j/issues/8907
Otherwise it's all passing.
